origins_workflow/src/Controller/ModerationStateController.php

```diff
@@ -102,9 +102,8 @@ private function transitionAllowed($entity, String $new_state) {
     // Check permissions of current user.
     $current_user = $this->currentUser();
     $transition_allowed = FALSE;
-    if (($current_state == 'draft') && ($new_state == 'published')) {
-      // Is this user allowed to use the 'quick publish' transition ?
-      if ($current_user->hasPermission('use editorial transition quick_publish')) {
+    if (($current_state == 'draft') && ($new_state == 'draft')) {
+      if ($current_user->hasPermission('use editorial transition create_new_draft')) {
         $transition_allowed = TRUE;
```

Comment:
This permission check was missing and was causing the watchdog error that was mentioned in the ticket